### PR TITLE
Goals Capture: Enable Goals Step i2 layout based on the feature flag

### LINKED ISSUE #64997: Goals Capture (i2): Create a new iteration step skeleton

### PR DESCRIPTION
**Pull request by phcp**
#### Proposed Changes

This PR enables Goals Step i2 layout based on the feature flag `signup/goals-step-2`:
- Creates a `GoalsCaptureContainer` component, which conditionally renders the **one column layout** if the flag is **disabled** and the **two columns layout** if the flag is **enabled**
- Applies the `GoalsCaptureContainer` on `GoalsStep` component, defining the `displayAllGoals` as true if the flag is enabled
  - Updates `useEffect` to not delete the **DIFM** and **Import** goals when the flag is true
- Restores the i2 styling in the _scss_ file, but now it's specified for a `.two-columns` parent selector
- Updates the `SelectGoals` component to render the UI based on the flag
  - If true: renders the "Select all that apply" hint on top of the goals list
  - If false: renders the bottom links for DIFM and Import

##### Testing with the flag enabled

1. Navigate to `http://calypso.localhost:3000/setup/goals?siteSlug=<YOUR-SITE>&flags=signup/goals-step-2`
2. Check if the **two columns layout** appears as expected
Screen Shot 2022-07-01 at 10 50 51

3. Check if the **mobile layout** appears as expected
Screen Shot 2022-07-01 at 11 03 06

4. Select **DIFM** and **Import** goals and refresh the page
5. Check if the selected goals persisted

##### Testing with the flag disabled

1. Navigate to `http://calypso.localhost:3000/setup/goals?siteSlug=<YOUR-SITE>`
2. Check if the **one column layout** appears as expected
Screen Shot 2022-07-01 at 11 01 32

3. Check if the **mobile layout** appears as expected
Screen Shot 2022-07-01 at 11 05 27

Related to #64997

### CODE REVIEW

**Comment by ivan-ottinger** [comment]
I took a first look at the code changes and it is all looking good. 👍🏼 Then I will test the PR on Monday.

**Review by ivan-ottinger — APPROVED** [review]
Works as expected. 👍🏼 I did not notice any regression.

**Testing with the flag enabled**

Check if the two columns layout appears as expected  
→ ✅ Two columns displayed correctly on the desktop view.

Check if the mobile layout appears as expected
→ ✅ The mobile layout displayed correctly, heading and subheading align left as expected.

Select DIFM and Import goals and refresh the page
Check if the selected goals persisted
→ ✅ Yes, both goals stay selected

---

**Testing with the flag disabled**

Check if the one column layout appears as expected
→ ✅ The one column layout displays correctly.

Check if the mobile layout appears as expected
→ ✅  The layout looks good on mobile as well.

**Comment by vishnugopal** [comment]
I noticed that import takes the highest priority here (which is different from before). Is there an expectation to redirect back from import to other flows once import is done?

File: client/landing/stepper/declarative-flow/internals/steps-repository/goals/index.tsx [hunk 1]
```diff
@@ -39,23 +38,26 @@ const GoalsStep: Step = ( { navigation } ) => {
 	const translate = useTranslate();
 	const welcomeText = translate( 'Welcome!' );
 	const whatAreYourGoalsText = translate( 'What are your goals?' );
-	const headerText = (
-		<>
-			{ welcomeText } <br /> { whatAreYourGoalsText }
-		</>
-	);
 	const subHeaderText = translate( 'Tell us what would you like to accomplish with your website.' );
+	const displayAllGoals = isEnabled( 'signup/goals-step-2' );
```
Comment (mashikag) [inline client/landing/stepper/declarative-flow/internals/steps-repository/goals/index.tsx]: Nit pick
This definition can go outside of the component

File: client/landing/stepper/declarative-flow/internals/steps-repository/goals/goals-capture-container.tsx [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+			<FormattedHeader
+				id={ 'goals-header' }
+				headerText={ whatAreYourGoalsText }
+				subHeaderText={ subHeaderText }
+			/>
+		}
+	/>
+);
+
+export const GoalsCaptureContainer: React.VFC< GoalsCaptureContainerProps > = ( {
```
Comment (mashikag) [inline client/landing/stepper/declarative-flow/internals/steps-repository/goals/goals-capture-container.tsx:65]: Oh cool, I was not aware of `VFC` type. That is cool to know! 👍

**Comment by mashikag** [comment]
The code looks good to me. I have tested the PR locally and it seems to work as expected. I did not spot any regressions for when the flag is disabled. 

@vishnugopal regarding questioning the flow, I think it is a right call. I realise we took this discussion off to slack. However I believe we can still proceed with merging the PR, and addressing changing the flow later if needed.

**Review by mashikag — APPROVED** [review]

File: client/landing/stepper/declarative-flow/internals/steps-repository/goals/index.tsx (hunk 1 above)
Comment (phcp) [reply client/landing/stepper/declarative-flow/internals/steps-repository/goals/index.tsx]: Fixed!

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7421430

Thank you @phcp for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7421430).

**Pull request merged**